### ISSUE TITLE
Feature Request: Keras method to extract the graph

### ISSUE THREAD

**Issue by SumNeuron** [tensorflow/tensorflow#21873]
Several parts of the Tensorflow documentation suggest to the reader to use the higher level APIs (Keras / Estimators) when there is no explicit need if the low level API. 

Of the high level APIs, there are some upwards mobility between the levels. For example, a custom Keras model can be converted into an Estimator. 

However, getting access to the lower level APIs from the higher level apis is not so simple.

It would be nice - at least for visualization purposes - to allow users to see the automatically generated graph produced by the higher level APIs. 

Taking a Keras model, converting it to an estimator, defining the `serving_input_receiver_fn`, and then exporting the estimator via `export_savedmodel` to see the graph is a bit convoluted for "high level" apis.

A subclassed Keras model, once compiled, could have a method `as_graph` to convert to the computational graph.

**Comment by facaiy** [comment]
I think [ tf.keras.callbacks.TensorBoard ](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard) might solve your problem, right?

**Comment by cy89** [comment]
@SumNeuron can you please comment on @facaiy 's suggestion?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by facaiy** [comment]
I'll close the issue, please be free to reopen it if new information becomes available.

**Comment by bionicles** [comment]
hi @facaiy -- it doesn't seem possible to add keras model graph to tensorboard if you're using GradientTape... all the instructions involve use of callbacks and model.fit. If this is possible, how? If not, can we add an easy way to submit keras model graph to tensorboard for visualization? 

tf.summary.FileWriter doesn't work in eager mode in 1.14... but tf.contrib is going away and tf.contrib.ResourceSummaryWriter cannot seemingly submit graph, and we can't get the graph session inside @tf.function to send to tensorboard. 

Please advise how do you add tensorboard keras graph when not using model.fit?

**Comment by facaiy** [comment]
@bionicles bion, can you file a new issue for it?
